Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -283,12 +307,26 @@ public static DartAnalysisServerService getInstance() {
     return ServiceManager.getService(DartAnalysisServerService.class);
   }
 
+  /**
+   * Returns {@link NavigationRegion}s for the given file.
+   * Empty if no regions.
+   */
+  @NotNull
+  public List<NavigationRegion> getNavigation(String file) {
+   List<NavigationRegion> sourceRegions = myNavigationData.get(file);
```

Comment:
DAS uses native slashes in file paths, but IDE always uses forward slashes in its model. So this code didn't work on Windows. I fixed it in a follow up commit.